packages/fuels-core/src/types/transaction_builders.rs

```diff
@@ -117,10 +111,23 @@ macro_rules! impl_tx_trait {
 
             fn fee_checked_from_tx(
                 &self,
-                params: &ConsensusParameters,
+                network_info: &NetworkInfo,
             ) -> Result<Option<TransactionFee>> {
-                let tx = self.clone().build()?.tx;
-                Ok(TransactionFee::checked_from_tx(params, &tx))
+                let mut tx = self.clone().build()?;
+
+                tx.precompute(&network_info.chain_id())?;
+
+                if tx.is_using_predicates() {
```

Comment:
Let's be kind to our future selves (and other devs) and put a comment around here explaining _why_ we need to precompute and why we're estimating predicates the way that we're doing here. This kind of thing makes a lot of sense to us now that we have the context, but it would be an absolute nightmare to reason about without this context.